src/universal/mutations/CreateProjectMutation.js

```diff
@@ -1,36 +1,90 @@
 import {commitMutation} from 'react-relay';
+import {handleProjectConnections} from 'universal/mutations/UpdateProjectMutation';
+import makeEmptyStr from 'universal/utils/draftjs/makeEmptyStr';
+import clientTempId from 'universal/utils/relay/clientTempId';
+import createProxyRecord from 'universal/utils/relay/createProxyRecord';
+import getOptimisticProjectEditor from 'universal/utils/relay/getOptimisticProjectEditor';
+import toTeamMemberId from 'universal/utils/relay/toTeamMemberId';
 
 const mutation = graphql`
-  mutation CreateProjectMutation($newProject: ProjectInput!, $area: AreaEnum) {
+  mutation CreateProjectMutation($newProject: CreateProjectInput!, $area: AreaEnum) {
     createProject(newProject: $newProject, area: $area) {
       project {
         id
         agendaId
         content
         createdAt
         createdBy
+        editors {
+          preferredName
+          userId
+        }
+        integration {
+          service
+          nameWithOwner
+          issueNumber
+        }
         sortOrder
         status
         tags
-        teamId
         teamMemberId
         updatedAt
         userId
+        teamId
+        team {
+          id
+          name
+        }
+        teamMember {
+          id
+          picture
+          preferredName
+        }
       }
     }
   }
 `;
 
 const CreateProjectMutation = (environment, newProject, area, onError, onCompleted) => {
-  // const {viewerId} = environment;
+  const {viewerId} = environment;
   return commitMutation(environment, {
     mutation,
-    variables: {area, newProject},
-    // updater: (store) => {
-    // },
-    // optimisticUpdater: (store) => {
-    // TODO add the team to the sidebar when we move teams to relay
-    // },
+    variables: {
+      area,
+      newProject
+    },
+    updater: (store) => {
+      const projectEditor = getOptimisticProjectEditor(store, newProject.userId);
+      const project = store.getRootField('createProject')
+        .getLinkedRecord('project')
+        .setLinkedRecords([projectEditor], 'editors');
+      handleProjectConnections(store, viewerId, project);
+    },
+    optimisticUpdater: (store) => {
+      const {teamId, userId} = newProject;
+      const teamMemberId = toTeamMemberId(teamId, userId);
+      const now = new Date().toJSON();
+      const optimisticProject = {
+        ...newProject,
+        id: clientTempId(teamId),
```

Comment:
this is the first of many passes where database IDs are not created on the client. that was done as a stopgap & i'd really like to move away from that since we no longer use UUIDs.